File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int close = 0;
        int open = 0;
        for (auto ch : s) {
            if (ch == '(') open++;
            else if (open > 0 && ch == ')') open--;
            else close++;
        }
        return open + close;
    }
};

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        opened = closed = 0

        for c in s:
            if c == '(':
                opened += 1
            elif opened > 0 and c == ')':
                opened -= 1
            else:
                closed += 1
            
        return opened + closed